commit c8774929aa29fe8d8aff0645b8f288869009c0c1
Author: Isaac Davis <isaac.davis@joyent.com>
Date:   2019-05-28T17:15:55+00:00 (4 months ago)
    
    TRITON-1707 `sdcadm post-setup prometheus` breaks if CNS allow_transfer metadata does not exist yet
    Reviewed by: Trent Mick <trentm@gmail.com>
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index ad091ac..7dd1539 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,11 @@
 
 # sdcadm Changelog
 
+## 1.27.3
+
+- TRITON-1707 `sdcadm post-setup prometheus` breaks if CNS allow\_transfer
+  metadata does not exist yet
+
 ## 1.27.2
 
 - TRITON-1706 'sdcadm post-setup dev-sample-data' will now ensure that sample
diff --git a/lib/procedures/add-allow-transfer.js b/lib/procedures/add-allow-transfer.js
index 8df90a0..ab02300 100644
--- a/lib/procedures/add-allow-transfer.js
+++ b/lib/procedures/add-allow-transfer.js
@@ -180,7 +180,9 @@ AddAllowTransferProcedure.prototype.execute =
                     return;
                 }
 
-                const existingIps = self.cnsSvc.metadata.allow_transfer;
+                const allow_transfer = self.cnsSvc.metadata.allow_transfer;
+                const existingIps =
+                    allow_transfer === undefined ? [] : allow_transfer;
                 const updatedIps = existingIps.concat(self.newIps);
                 sapi.updateService(self.cnsSvc.uuid, {
                     metadata: {
diff --git a/package.json b/package.json
index 7f73718..49029fa 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a Triton Data Center",
-  "version": "1.27.2",
+  "version": "1.27.3",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
